### PR TITLE
Add Dockerfile to support docker builds.

### PR DESCRIPTION
**Pull request by xlab**
Hi, thanks for this great tool. I'd like to make it docker-enabled, so I can produce cpython artefacts for `Linux x86_64` while having `Darwin x86_64` as my primary working environment.

See the `Dockerfile`. Now this repo can be hosted on [docker hub](https://hub.docker.com) or built locally:
```
$ docker build -t go-python/gopy .
```

**Usage example:**
```
$ cd ~/go/src/github.com/go-python/gopy/_examples/hi
$ docker run -it --rm -v `pwd`:/go/src/in -v `pwd`:/out go-python/gopy app bind -output=/out in
$ file hi.so
hi.so: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, not stripped
```

Works fine without any external deps ever, on any host capable of docker.

### CODE REVIEW

**Comment by sbinet** [comment]
Thanks.
Could you also add the nice little doc snippet in the `README` as well ?

**Comment by xlab** [comment]
Sure. By the way, are you going to publish an Automated Build on [Docker Hub](https://hub.docker.com)?

I just did so with the forked one: https://hub.docker.com/r/xlab/gopy/
It can be used via `docker pull xlab/gopy`, but idiomatically this should be `docker pull go-python/gopy`, that's up to you.

**Comment by sbinet** [comment]
I have just created an organization on the hub:
 https://hub.docker.com/u/gopy/dashboard/

I had to use `gopy` instead of `go-python` (which was an invalid name...)

could you update the instructions accordingly? (thanks again)

File: README.md [hunk 1]
```diff
@@ -146,6 +146,23 @@ PASS
 ok  	github.com/go-python/gopy	2.135s
 ```
 
+## Binding generation using Docker (for cross-platform builds)
+
+```
+$ cd github.com/go-python/gopy/_examples/hi
+$ docker run -it --rm -v `pwd`:/go/src/in -v `pwd`:/out xlab/gopy app bind -output=/out in
```
Comment (sbinet) [inline README.md]: not sure the `-it` is needed in this particular case.

Comment (sbinet) [inline README.md]: also:
`s|xlab/gopy|gopy/gopy|`

**Comment by sbinet** [comment]
needs go-python/license#2

**Comment by sbinet** [comment]
LGTM

**Comment by sbinet** [comment]
merged bc0afa9

**Comment by xlab** [comment]
Great, thanks! You probably should trigger the first build of `master` via https://hub.docker.com/r/gopy/gopy/~/settings/automated-builds/
So I can use `docker pull gopy/gopy`.

**Comment by xlab** [comment]
:(

**Comment by sbinet** [comment]
Apologies. I have been swamped with the last rush before x-mas.
Also, I was a bit lost with the new web interface.
If you want, I can make you part of the organization on the hub so you can do your magic.

**Comment by xlab** [comment]
Sure thing, I can maintain. https://hub.docker.com/u/xlab/

**Comment by xlab** [comment]
I've been notified about `devs team` group membership, however I still can't trigger anything.
Btw, this may help:  — the only thing you need to do is to trigger a build.

**Comment by sbinet** [comment]
ok, I think we are all set now.
(I was confused because, before, you could create an image spec, and then make it an automated build, now it's the other way around. anyways...)

**Comment by xlab** [comment]
Great, thanks!

```
$ docker pull gopy/gopy
Using default tag: latest
latest: Pulling from gopy/gopy

Digest: sha256:19cafa11878b437e20792309f459a6df28f74b2987cc92796ccb6b75b50f26b7
Status: Downloaded newer image for gopy/gopy:latest
```

**Pull request closed without merging**